### PR TITLE
[4.0] Stop TinyMCE autochanging skins when new skins are uploaded

### PR DESCRIPTION
**Pull request by SharkyKZ**
Redo of #22097 .

### Summary of Changes

Currently, we store TinyMCE skin setting as a numeric value provided by `glob()`. This can cause the skin to automatically change when uploading a new skin.
This removes custom `JFormFieldSkins` field and replaces it with standard folder list field in form.

### Testing Instructions

Edit TinyMCE plugin. In your set (Set 0 by default) select a skin. Save the plugin.
Upload a skin with name that alphabetically orders before your selected skin.
Edit an article.
You should see the new skin being used or the editor not appearing at all.
Remove the custom skin.
Apply patch.
Save plugin configuration again.
Upload the skin again.
Edit an article.
Now your selected skin should be used.

### Documentation Changes Required

`JFormFieldSkins` is removed.

### CODE REVIEW

**Comment by jaydevingawale** [comment]
I have tested this item :red_circle: unsuccessfully on 16dea33630d8460f36c53811d3a0b81d28597ecf<br /><br />It still shows the blank page<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25493">issues.joomla.org/tracker/joomla-cms/25493</a>.</sub>

**Comment by softforge** [comment]
This does indeed fail to stop the plugin change and to stop the editor breaking

**Comment by softforge** [comment]
I have tested this item :red_circle: unsuccessfully on 16dea33630d8460f36c53811d3a0b81d28597ecf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25493">issues.joomla.org/tracker/joomla-cms/25493</a>.</sub>

File: plugins/editors/tinymce/form/setoptions.xml [hunk 1]
```diff
@@ -19,14 +19,22 @@
 
     <field
         name="skin"
-        type="skins"
+        type="folderlist"
         label="PLG_TINY_FIELD_SKIN_LABEL"
+		directory="media/vendor/tinymce/skins/ui"
```
Comment (Quy) [inline plugins/editors/tinymce/form/setoptions.xml:24]: Reduce indentation.

Comment (SharkyKZ) [reply plugins/editors/tinymce/form/setoptions.xml:24]: My indentation is correct. It's the rest of XML that uses spaces. I'll make a separate PR.

Comment (SharkyKZ) [reply plugins/editors/tinymce/form/setoptions.xml:24]: See #27577.

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on bcc2758b490913a47899cdc8bc64a0aebf4c986e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25493">issues.joomla.org/tracker/joomla-cms/25493</a>.</sub>

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 3075d410a7e2ec7823108d5eb0635bda2158ff9f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25493">issues.joomla.org/tracker/joomla-cms/25493</a>.</sub>

**Comment by jwaisner** [comment]
@flyingwombats @jaydevingawale Please retest this.

**Comment by wilsonge** [comment]
Should we also set default values in the XML (given we already have them in the PHP?). Other than that LGTM

**Comment by SharkyKZ** [comment]
I intentionally left this part the way it was.

I guess we could add the ` - Use Default - ` option with empty value. This way, if selected theme is deleted, the field will use empty value rather than the first option alphabetically. And we'll fall back to `oxide` in the code.

**Comment by wilsonge** [comment]
LGTM!

**Pull request merged**